### PR TITLE
REF: finish setting pytables IndexCol attrs in constructor

### PR DESCRIPTION
**Pull request by jbrockmendel**
we're not all the way done, but this is a big milestone

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -2490,8 +2475,7 @@ def get_atom_timedelta64(cls, shape):
 class GenericDataIndexableCol(DataIndexableCol):
```
Comment (WillAyd) [inline pandas/io/pytables.py]: Can we just blow this class away altogether now?

Comment (jbrockmendel) [reply pandas/io/pytables.py]: worth a shot

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -3469,18 +3459,29 @@ def f(i, c):
...
             adj_name = _maybe_adjust_name(c, self.version)
+
+            # TODO: why kind_attr here?
+            values = getattr(table_attrs, f"{adj_name}_kind", None)
+            dtype = getattr(table_attrs, f"{adj_name}_dtype", None)
+            kind = _dtype_to_kind(dtype)
+
             md = self.read_metadata(c)
-            meta = "category" if md is not None else None
+            # TODO: figure out why these two versions of `meta` dont always match.
```
Comment (jreback) [inline pandas/io/pytables.py]: what is going on here?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/pytables.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: So ever place where we set `self.meta` it is equivalent to `meta = "category" if md is not None else none`, but in this one place where we read it from the file, there is one test case where it doesnt match, and i dont know why.

if/when that is resolved, `meta` can be made a property

**Comment by jreback** [comment]
can you rebase

**Comment by jbrockmendel** [comment]
rebased+green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
k thanks

**Pull request merged**